x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotRetentionConfiguration.java

```diff
@@ -149,25 +167,30 @@ public Integer getMaximumSnapshotCount() {
             // if we haven't hit the minimum then we need to keep the snapshot regardless of
             // expiration time
             if (this.minimumSnapshotCount != null) {
-                if (allSnapshots.size() <= this.minimumSnapshotCount) {
-                    logger.trace("[{}]: INELIGIBLE as there are {} snapshots and {} minimum snapshots needed",
-                        snapName, snapCount, this.minimumSnapshotCount);
+                if (nonFailedSnapshotCount <= this.minimumSnapshotCount && SnapshotState.FAILED.equals(si.state()) == false) {
```

Comment:
Can you split this into two `if` statements so we can have logging if they are below the minimum but it *is* a failure? I think it'd be easier to reason about then too